service/assessment/assessment.go

```diff
@@ -52,9 +52,9 @@ Service is an implementation of the Clouditor Assessment service. It should not
 NewService constructor should be used. It implements the AssessmentServer interface
 */
 type Service struct {
-	// ResultHook is a hook function that can be used if one wants to be
+	// resultHook is a hook function that can be used if one wants to be
 	// informed about each assessment result
-	ResultHook []func(result *assessment.AssessmentResult, err error)
+	resultHook []func(result *assessment.AssessmentResult, err error)
```

Comment:
Let's call it resultHooks. I would also suggest declaring a `ResultHookFunc` type, to make it easier for users which function signature to implement.